modules/objdetect/src/qrcode_encoder.cpp

```diff
@@ -1334,9 +1334,13 @@ class QRCodeDecoderImpl : public QRCodeDecoder {
                 actualBits = 8;
             }
             if (bits) {
-                val |= data[idx] >> (actualBits - bits);
-                actualBits -= bits;
-                data[idx] &= 255 >> (8 - actualBits);
+                if (idx >= data.size()) {
+                    // CV_LOG_WARNING(NULL, "Bitstream read overrun - some bits may be skipped");
```

Comment:
Error should be raise instead of providing of wrong result.

Need to modify check on the line 1328 with current `CV_Assert(idx < data.size());`

Reply:
I propose assert here instead of condition.

Reply:
Just here is nit enough.

`while` loop above has the same issue with buffer access.

Reply:
If we add assertion here, our tests will fail because there are cases where more bits are being read. We need to fix these cases first.